### PR TITLE
Completes possible list of kind/ labels on PRs/Issues

### PR DESCRIPTION
**Pull request by guineveresaenger**
Also links to test-infra's label reference, which is more complete, but not up to date either.

### CODE REVIEW

**Comment by guineveresaenger** [comment]
This is somewhat related to the effort for a better pull request template in k/k, which will link to this document, so the content should be more complete: https://github.com/kubernetes/kubernetes/pull/69837

File: contributors/devel/release.md [hunk 1]
```diff
@@ -300,20 +302,21 @@ of the issue.
...
   - moved out of milestone more aggressively than `priority/important-soon`
 
-### Issue Kind Label
+### Issue/PR Kind Label
 
 The issue kind is used to help identify the types of changes going
 into the release over time.  This may allow the release team to
 develop a better understanding of what sorts of issues we would
-miss with a faster release cadence.
+miss with a faster release cadence. These labels may also cause escalation to the correct SIG GitHub team.
```
Comment (cblecker) [inline contributors/devel/release.md]: nit: wrap 80

Comment (cblecker) [reply contributors/devel/release.md]: also, is this true anymore? I don't think it is as the milestone maintainer is dead

File: contributors/devel/release.md [hunk 2]
```diff
@@ -300,20 +302,21 @@ of the issue.
...
 
-This may also be used to escalate to the correct SIG GitHub team.
-For release targeted issues one of the follow issue types must be
-set (additional may also be set):
+For release targeted issues, including pull requests, one of the follow issue kind labels must be
```
Comment (cblecker) [inline contributors/devel/release.md]: nit: wrap 80

File: contributors/devel/release.md [hunk 3]
```diff
@@ -300,20 +302,21 @@ of the issue.
...
 
 - `kind/bug`: Fixes a newly discovered bug.
-  - were not known issues at the start of the development period.
 - `kind/feature`: New functionality.
 - `kind/cleanup`: Adding tests, refactoring, fixing old bugs.
 - `kind/failing-test`: CI test case is failing consistently.
 - `kind/flake`: CI test case is showing intermittent failures.
+- `kind/api-change`: Adds, removes, or changes an API
+- `kind/documentation`: Adds documentation
+- `kind/design`: Related to design
```
Comment (cblecker) [inline contributors/devel/release.md]: nit: alpha sort this list?

File: contributors/devel/release.md (hunk 1 above)
Comment (guineveresaenger) [reply contributors/devel/release.md]: I just re-edited what was there before, I'm happy to delete it altogether. It was extremely unclear to me what the original sentence meant.

File: contributors/devel/release.md [hunk 4]
```diff
@@ -300,20 +302,21 @@ of the issue.
   - even less urgent / critical than `priority/important-soon`
   - moved out of milestone more aggressively than `priority/important-soon`
 
-### Issue Kind Label
+### Issue/PR Kind Label
```
Comment (justaugustus) [inline contributors/devel/release.md]: super nit: space between issue and PR

**Comment by justaugustus** [comment]
Mentioned in https://github.com/kubernetes/kubernetes/pull/69837
> > It... turns out test-infra's label doc is not complete either, it's missing some area labels I know exist.
> 
> @guineveresaenger -- which labels are missing from https://git.k8s.io/test-infra/label_sync/labels.md ? That list is generated from a script, so it should absolutely be correct. If you see something that looks off, let us know.

Let's get some clarification on whether something is actually off with the test-infra labels list, before merging this.

Otherwise, lgtm.

File: contributors/devel/release.md (hunk 4 above)
Comment (guineveresaenger) [reply contributors/devel/release.md]: I didn't think so? https://www.grammarly.com/blog/slash/

**Comment by spiffxp** [comment]
/hold cancel
*mumbles something about duplicate label descriptions inevitably falling out of date*

**Pull request merged**